drivers/flash/nrf_qspi_nor.c

```diff
@@ -476,17 +476,42 @@ static inline int qspi_nor_read_id(struct device *dev,
 	return 0;
 }
 
+#if (CONFIG_HEAP_MEM_POOL_SIZE > 0)
+u8_t *rx_ram_buf;
+u8_t *tx_ram_buf;
+#endif
+
 static int qspi_nor_read(struct device *dev, off_t addr, void *dest,
 			 size_t size)
 {
+	int ret;
+
 	if (!dest) {
 		return -EINVAL;
 	}
+#if (CONFIG_HEAP_MEM_POOL_SIZE > 0)
+	rx_ram_buf = NULL;
+	void *tmp_dest = dest;
+	size_t var_size = size;
+
+	if (((uint32_t)dest < CONFIG_SRAM_BASE_ADDRESS)
+	   || (size < sizeof(uint32_t))) {
```

Comment:
if (((uint32_t)dest < CONFIG_SRAM_BASE_ADDRESS) ||
	     (size < sizeof(uint32_t))) {
Maybe move logic operator? I find it more readable.